vmm/src/signal_handler.rs

```diff
@@ -172,26 +178,33 @@ mod tests {
             // Call the blacklisted `SYS_mkdirat`.
             unsafe { syscall(libc::SYS_mkdirat, "/foo/bar\0") };
 
-            assert!(cpu_count() > 0);
-
-            // Kcov somehow messes with our handler getting the SIGSYS signal when a bad syscall
-            // is caught, so the following assertion no longer holds. Ideally, we'd have a surefire
-            // way of either preventing this behaviour, or detecting for certain whether this test is
-            // run by kcov or not. The best we could do so far is to look at the perceived number of
-            // available CPUs. Kcov seems to make a single CPU available to the process running the
-            // tests, so we use this as an heuristic to decide if we check the assertion.
-            if cpu_count() > 1 {
-                // The signal handler should let the program continue during unit tests.
-                assert_eq!(METRICS.seccomp.num_faults.count(), 1);
+            // Call SIGBUS signal handler.
+            assert_eq!(METRICS.signals.sigbus.count(), 0);
+            unsafe {
+                syscall(libc::SYS_kill, process::id(), SIGBUS);
             }
 
-            // Call SIGBUS signal handler. We are not testing that the SIGBUS handler was executed
-            // successfully.
-            // Tracking issue: https://github.com/firecracker-microvm/firecracker/issues/1141
+            // Call SIGSEGV signal handler.
```

Comment:
We are still not checking that SIGBUS left the process alive because if the process exists the asserts at lines 196, 207 and 208 will not be executed anyway hence you cannot know if they passed or failed.

Reply:
Well, if any of the previous `assert!`s fail, it's still a sign of something wrong. Normal functioning will let the `assert!`s pass.